pynamodb/connection/base.py

```diff
@@ -866,20 +867,26 @@ def update_item(self,
         if not attribute_updates:
             raise ValueError("{0} cannot be empty".format(ATTR_UPDATES))
 
-        operation_kwargs[ATTR_UPDATES] = {}
-        for key, update in attribute_updates.items():
-            value = update.get(VALUE)
-            attr_type, value = self.parse_attribute(value, return_type=True)
+        update_expression = Update()
+        # We sort the keys here for determinism. This is mostly done to simplify testing.
+        for key in sorted(attribute_updates.keys()):
+            update = attribute_updates[key]
             action = update.get(ACTION)
-            if attr_type is None and action is not None and action.upper() != DELETE:
-                attr_type = self.get_attribute_type(table_name, key, value)
             if action not in ATTR_UPDATE_ACTIONS:
                 raise ValueError("{0} must be one of {1}".format(ACTION, ATTR_UPDATE_ACTIONS))
-            operation_kwargs[ATTR_UPDATES][key] = {
-                ACTION: action,
-            }
-            if action.upper() != DELETE:
-                operation_kwargs[ATTR_UPDATES][key][VALUE] = {attr_type: value}
+            value = update.get(VALUE)
+            attr_type, value = self.parse_attribute(value, return_type=True)
+            if attr_type is None and action != DELETE:
```

Comment:
don't need the `.upper()` anymore?

Reply:
nope, the allowed values are all upper-case